File: Medium/Subsets/subsets.py
#User function Template for python3

class Solution:
    def subsets(self, l):
        n=len(l)
        p=1<<(len(l))
        k1=[]
        for i in range(p):
           k=[]
           for j in range(n):
               if(i&(1<<j)):
                    k.append(l[j])
           k1.append(k)
        return sorted(k1)
               
                #code here


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    test_cases = int(input())
    for cases in range(test_cases):
        n = int(input())
        A = list(map(int,input().strip().split()))
        
        ob=Solution()
        result =ob.subsets(A)
        
        for i in range(len(result)):
            for j in range(len(result[i])):
                print(result[i][j],end=" ")
                
            print()
            
            

# } Driver Code Ends

File: Medium/Subsets/subsets.cpp
//{ Driver Code Starts
//Initial Template for C++

#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
//User function Template for C++

class Solution
{
    public:
    vector<vector<int>>ans;
    void fun(vector<int>& A,vector<int>& curr,int i)
    {
        if(i==A.size())
        {
            ans.push_back(curr);
            return ;
        }
        fun(A,curr,i+1);
        curr.push_back(A[i]);
        fun(A,curr,i+1);
        curr.pop_back();
    }
    vector<vector<int> > subsets(vector<int>& A)
    {
        //code here
        vector<int>v;
        vector<int>curr;
        fun(A,curr,0);
        sort(ans.begin(),ans.end());
        return ans;
    }
};

//{ Driver Code Starts.

int main()
{
	int t;
	cin >> t;

	while (t--)
	{
		int n, x;
		cin >> n;

		vector<int> array;
		for (int i = 0; i < n; i++)
		{
			cin >> x;
			array.push_back(x);
		}
        
        
        Solution ob;
		vector<vector<int> > res = ob.subsets(array);

		// Print result
		for (int i = 0; i < res.size(); i++) {
			for (int j = 0; j < res[i].size(); j++)
				cout << res[i][j] << " ";
			cout << endl;
		}

		
	}

	return 0;
}
// } Driver Code Ends